recipes/eigen/all/conanfile.py

```diff
@@ -10,19 +12,31 @@ class EigenConan(ConanFile):
                    numerical solvers, and related algorithms."
     license = ("MPL-2.0", "LGPL-3.0-or-later")
     topics = ("eigen", "algebra", "linear-algebra", "vector", "numerical")
-    settings = "os", "compiler", "build_type", "arch"
-    options = {"MPL2_only": [True, False]}
-    default_options = {"MPL2_only": False}
-    exports_sources = ["patches/*"]
-    no_copy_source = True
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "MPL2_only": [True, False],
```

Comment:
this option should not have been accepted in https://github.com/conan-io/conan-center-index/pull/4342, it's a pure interface definition without more logic like adding dependencies.

Reply:
well, we have such options, which don't affect on package contents, but just add some additional preprocessor definitions to the package_info, it's fine

Reply:
I think it makes recipes confusing and adds unnecessary complexity. Outside of conan, nobody would expect to see definition injected by another tool.